src/clj/commiteth/db/seed.clj

```diff
@@ -0,0 +1,193 @@
+
+(s/def ::hook_secret string?)
+
+(s/def ::owner_avatar_url ::avatar_url)
+
+(s/def ::repository (s/keys :req-un [::repo_id
+                                     ::user_id
+                                     ::owner
+                                     ::repo
+                                     ::hook_id
+                                     ::state
+                                     ::hook_secret
+                                     ::owner_avatar_url]))
+
+(s/def ::repo_id non-nil-pos-int)
+(s/def ::issue_id non-nil-pos-int)
+(s/def ::issue_number non-nil-pos-int)
+(def issue-nouns  ["compiler" "solc" "re-frame" "wallet" "ethereum" "ropsten network"])
+(def issue-verbs ["broke" "fix" "deploy" "not working" "hanging on page load" "on fire"])
+(defn gen-issue-title []
+  "generates a random issue title"
+  (str (rand-nth issue-nouns) " " (rand-nth issue-verbs)))
+
+(s/def ::title (s/with-gen string?
+                 #(s/gen (conj #{} (gen-issue-title)))))
+
+(defn generate-eth-value [max-eth-bounty precision]
+  (-> max-eth-bounty
+      rand
+      bigdec
+      (.setScale precision java.math.RoundingMode/HALF_EVEN)
+      float))
+
+(s/def ::balance_eth (s/with-gen float?
+                       #(s/gen (conj #{} (generate-eth-value 3 4)))))
+
+(defn generate-usd-value [max-usd]
+  (float (Math/floor (rand max-usd))))
+
+(s/def ::value_usd (s/with-gen float?
+                     #(s/gen (conj #{} (generate-usd-value 300)))))
+
+(s/def ::commit_sha string?)
+(s/def ::issue_number non-nil-pos-int)
+
+(s/def ::issue (s/keys :req-un [::repo_id
+                                ::issue_id
+                                ::title
+                                ::balance_eth
+                                ::value_usd
+                                ::issue_number
+                                ::commit_sha]))
+
+(s/def ::pr_id non-nil-pos-int)
+(s/def ::pr_number non-nil-pos-int)
+
+(s/def ::pull_request (s/keys :req-un [::pr_id
+                                       ::pr_number
+                                       ::title
+                                       ::repo_id
+                                       ::user_id
+                                       ::commit_sha
+                                       ::issue_number
+                                       ::state
+                                       ::issue_id]))
+
+(defn gen-spec [spec]
+  "utility used by specmonstah to generate sample data"
+  (gen/generate (s/gen spec)))
+
+(def relations
+  (rs/expand-relation-template
+   {::user         [{}]
+    ::repository   [{:user_id [::user :id]
+                     :owner [::user :name]}]
+    ::issue        [{:repo_id [::repository :repo_id]}]
+    ::pull_request [{:user_id  [::user :id]
+                     :repo_id  [::repository :repo_id]
+                     :issue_id [::issue :issue_id]}]}))
+
+(def pull-request-result (rs/gen-tree gen-spec relations [::pull_request]))
+
+(def inserted-records (atom []))
```

Comment:
As this atom is more part of the script entry point rather than the actual seeding code it might be better to `let` it inside the `seed-postgres` function. Similar points apply to `insert!` I guess.

Reply:
nice points to raise, thanks!